### DISCUSSION TITLE
[Compiler] Trap node not identified in module level

### DISCUSSION THREAD

**Discussion by gayalkuruppu** [ballerina-platform/ballerina-lang#34008] [General]
[Updated]

In module level, the expression keywords such as `trap`, `check`, and `checkpanic` are identified as the `ModulePartNode`. related: https://github.com/ballerina-platform/ballerina-lang/issues/33833 

Consider the following scenario:
![Screenshot 2021-11-26 at 09 40 46](https://user-images.githubusercontent.com/46120162/143525637-f10965d1-a056-4e98-b035-5e13d989d5e4.png)

Even though the code line with `trap` is not correct syntactically, we provide completions here and then give code actions to create a variable afterwards. Which makes the code line syntactically correct. This approach is used to improve the user experience and guide/ help the user to write code easily/correct even though started with an incorrect line. 

https://user-images.githubusercontent.com/46120162/143526934-f53895dc-5a89-464f-a659-bed460b5c097.mov

But in the module part, for expression related keywords we do not get completions due to this limitation of parser. So our UX end up like this in the module level.

![Screenshot 2021-11-26 at 10 03 48](https://user-images.githubusercontent.com/46120162/143527246-17ba2214-e94e-46ec-ae80-d0ceec0d5929.png)

It will be good if we can give the same UX in the module level as well. So can we fix this in order to identify these nodes as their respective matching nodes instead of `ModulePartNode`?

@nadeeshaan @lochana-chathura @rdulmina

**Comment by lochana-chathura** [comment]
@gayalkuruppu Can you provide a code sample?

**Reply by gayalkuruppu** [reply]
@lochana-chathura please refer to this case https://github.com/ballerina-platform/ballerina-lang/issues/33833

**Reply by lochana-chathura** [reply]
At the module-level parser can't identify expressions without their lhs assignment. 
e.g. 
```bal
var myvar = trap <cursor> // can identify
trap <cursor> // cannot identify
```

For the second case, it is the same reason as this issue #27268.

**Reply by nadeeshaan** [reply]
In order to add some more context to this,

At the block node level, we can identify the particular expression, since we support expression statement as per the grammar and in the sense, in the parser. But at the top level, we don't do this and hence we cannot identify the valid context.
